test/unittest/xmlparser/XMLElementParserTests.cpp

```diff
@@ -366,6 +469,63 @@ TEST_F(XMLParserTests, getXMLLocatorUDPv6)
     EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
 }
 
+/*
+ * This test checks the parsing of a <udpv6> element from a list of locators, using DNS resolution.
+ * 1. Correct parsing of a valid element (with address given by domain).
+ */
+TEST_F(XMLParserTests, getXMLLocatorDNSUDPv6)
+{
+
+    uint8_t ident = 1;
+    LocatorList_t list;
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+
+    // Parametrized XML
+    const char* xml_p =
+            "\
+            <unicastLocatorList>\
+                <locator>\
+                    <udpv6>\
+                        <port>%s</port>\
+                        <address>%s</address>\
+                        %s\
+                    </udpv6>\
+                </locator>\
+            </unicastLocatorList>\
+            ";
+    constexpr size_t xml_len {500};
+    char xml[xml_len];
+
+    // Valid XML
+    snprintf(xml, xml_len, xml_p, "8844", "www.acme.com.test", "");
+    ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+    titleElement = xml_doc.RootElement();
+    EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::getXMLLocatorList_wrapper(titleElement, list, ident));
+    EXPECT_EQ(list.begin()->port, 8844u);
+
+    // <address>
+    Locator_t expected_locator; // auxiliar locator to check the address, disregard port
+    IPLocator::createLocator(LOCATOR_KIND_UDPv6, "2a00:1450:400e:803::2004", 8844, expected_locator);
+    EXPECT_EQ(list.begin()->address[0], expected_locator.address[0]);
+    EXPECT_EQ(list.begin()->address[1], expected_locator.address[1]);
+    EXPECT_EQ(list.begin()->address[2], expected_locator.address[2]);
+    EXPECT_EQ(list.begin()->address[3], expected_locator.address[3]);
+    EXPECT_EQ(list.begin()->address[4], expected_locator.address[4]);
+    EXPECT_EQ(list.begin()->address[5], expected_locator.address[5]);
+    EXPECT_EQ(list.begin()->address[6], expected_locator.address[6]);
+    EXPECT_EQ(list.begin()->address[7], expected_locator.address[7]);
+    EXPECT_EQ(list.begin()->address[8], expected_locator.address[8]);
+    EXPECT_EQ(list.begin()->address[9], expected_locator.address[9]);
+    EXPECT_EQ(list.begin()->address[10], expected_locator.address[10]);
+    EXPECT_EQ(list.begin()->address[11], expected_locator.address[11]);
+    EXPECT_EQ(list.begin()->address[12], expected_locator.address[12]);
+    EXPECT_EQ(list.begin()->address[13], expected_locator.address[13]);
+    EXPECT_EQ(list.begin()->address[14], expected_locator.address[14]);
+    EXPECT_EQ(list.begin()->address[15], expected_locator.address[15]);
+    EXPECT_EQ(list.begin()->kind, LOCATOR_KIND_UDPv6);
```

Comment:
```suggestion
        EXPECT_EQ(*list.begin(), expected_locator);
```
Maybe we can use `==` operator of Locator_t to compare.